src/ray/raylet/scheduling_resources.cc

```diff
@@ -170,16 +261,34 @@ ResourceIds::ResourceIds(double resource_quantity) {
   for (int64_t i = 0; i < whole_quantity; ++i) {
     whole_ids_.push_back(i);
   }
+  greatest_id_ = resource_quantity;
+  total_capacity_ = TotalQuantity();
```

Comment:
If we're adding an extra `total_capacity_` field, then do we need the `TotalQuantity` method? Can't we just have `TotalQuantity` return `total_capacity_`?

Reply:
I think the name `TotalQuantity()` is a bit misleading, since it returns the number of resources currently in the `whole_ids_` vector. The `whole_ids_` vector size keeps changing when resources are acquired or released, thus `TotalQuantity()` reflects the available resources. Thus I added `total_capacity_` to keep a track of the total count of resources in `ResourceIds`.